tests/storage/export.go

```diff
@@ -1211,14 +1274,54 @@ var _ = SIGDescribe("Export", func() {
 		token := createExportTokenSecret(snapshot.Name, snapshot.Namespace)
 		export := createVMSnapshotExportObject(snapshot.Name, snapshot.Namespace, token)
 		Expect(export).ToNot(BeNil())
-		Eventually(func() exportv1.VirtualMachineExportPhase {
-			export, err = virtClient.VirtualMachineExport(export.Namespace).Get(context.Background(), export.Name, metav1.GetOptions{})
-			Expect(err).ToNot(HaveOccurred())
-			if export.Status == nil {
+		waitForExportPhase(export, exportv1.Skipped)
+	})
+
+	It("should report export pending if VM is running, and start the VM export if the VM is not running", func() {
+		sc, exists := libstorage.GetRWOFileSystemStorageClass()
+		if !exists {
+			Skip("Skip test when Filesystem storage is not present")
+		}
+		vm := tests.NewRandomVMWithDataVolumeWithRegistryImport(
+			cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskAlpine),
+			util.NamespaceTestDefault,
+			sc,
+			k8sv1.ReadWriteOnce)
+		vm.Spec.Running = pointer.BoolPtr(true)
+		vm = createVM(vm)
+		Eventually(func() virtv1.VirtualMachineInstancePhase {
+			vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(vm.Name, &metav1.GetOptions{})
+			if errors.IsNotFound(err) {
 				return ""
 			}
-			return export.Status.Phase
-		}, 30*time.Second, time.Second).Should(Equal(exportv1.Skipped))
+			Expect(err).ToNot(HaveOccurred())
+			return vmi.Status.Phase
+		}, 180*time.Second, time.Second).Should(Equal(virtv1.Running))
+		// For testing the token is the name of the source VM.
+		token := createExportTokenSecret(vm.Name, vm.Namespace)
+		export := createVMExportObject(vm.Name, vm.Namespace, token)
+		Expect(export).ToNot(BeNil())
+		waitForExportPhase(export, exportv1.Pending)
+		waitForExportCondition(export, pvcInUseCondition, "export should report pvc in use")
+
+		By("Stopping VM, we should get the export ready eventually")
+		vm = stopVM(vm)
+		export = waitForReadyExport(export)
+		verifyKubevirtInternal(export, export.Name, export.Namespace, vm.Spec.Template.Spec.Volumes[0].DataVolume.Name)
+		By("Starting VM, the export should return to pending")
+		vm = startVM(vm)
+		waitForExportPhase(export, exportv1.Pending)
+		waitForExportCondition(export, pvcInUseCondition, "export should report pvc in use")
+	})
+
+	It("should mark the status phase skipped on VM without volumes", func() {
+		vm := tests.NewRandomVMWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskCirros))
+		vm = createVM(vm)
+		// For testing the token is the name of the source VM.
+		token := createExportTokenSecret(vm.Name, vm.Namespace)
+		export := createVMExportObject(vm.Name, vm.Namespace, token)
+		Expect(export).ToNot(BeNil())
+		waitForExportPhase(export, exportv1.Skipped)
 	})
```

Comment:
Would like an additional test case for export running, vm started, export cancelled